client/boptest.py

```diff
@@ -0,0 +1,262 @@
+        mongodb = mongo_client['admin']
+        recs = mongodb.recs
+        recs.update_one({"_id": site_ref}, {"$set": {"rec.simStatus": "s:Running"}}, False)
+
+        config = { 'fmupath': fmupath, 'step': 60 }
+        
+                
+        tc = testcase.TestCase(config)
+        u = {}
+        #run the FMU simulation
+        kstep=0
+        stop = False
+        simtime = 0
+        while simtime < 120 and not stop:
+            y_output = tc.advance(u)
+            simtime = tc.final_time
+            output_time_string = 's:%s' %(simtime)
+            recs.update_one( {"_id": site_ref}, { "$set": {"rec.datetime": output_time_string, "rec.simStatus":"s:Running"} } )
+
+        return 0
+        
+    
+    def get_tag_data(self, tag_filepath):
+        '''
+        Purpose: retrieving the haystack tagged data, 
+              which are model-exchange variables in FMU. 
+        Inputs:  a json file containing all the tagged data.
+        Returns: a list for all the tagged data, with the tagging properties
+              like id, dis, site_ref, etc...
+        '''
+        with open( tag_filepath ) as json_data:
+            tag_data = json.load(json_data)
+     
+        return tag_data
+
+
+    #This is the function to create tags
+    def create_DisToID_dictionary(self, tag_filepath):
+        '''
+        Purpose: matching the haystack display-name and IDs
+        Inputs:  a json file containing all the tagged data
+        Returns: a dictionary mathching all display-names and IDs
+             a dictionary matching all inputs and IDs
+             a dictionary matching all outputs and IDs 
+             a dictionary matching all IDs and display-names 
+        '''
+        dis_and_ID={}
+        inputs_and_ID = {}
+        outputs_and_ID = {}
+        id_and_dis={}
+
+        tag_data = get_tag_data(tag_filepath)
+
+        for point in tag_data:
+            var_name = point['dis'].replace('s:','')
+            var_id = point['id'].replace('r:','')
+
+            dis_and_ID[var_name] = var_id
+            id_and_dis[var_id] = var_name
+
+            if 'writable' in point.keys():
+                #it means it is a input variable.
+                #clean the var-name, discarding: ':input','s:','r:'
+                #input_var = var_name.replace(':input','')
+                #input_var = input_var.replace('s:','')
+                inputs_and_ID[var_name] = var_id
+
+            if 'writable' not in point.keys() and 'point' in point.keys():
+                #it means it is an output variable, plus not sitetag.
+                #clean the var-name, discarding: ':output','s:','r:'
+                #output_var = var_name.replace(':output','')
+                #output_var = output_var.replace('s:','')
+                outputs_and_ID[var_name] = var_id
+
+        return (dis_and_ID, inputs_and_ID, outputs_and_ID, id_and_dis)
+
+
+
+    # Stop a simulation for model identified by id
+    def stop(self, id):
```

Comment:
Like start, send an http request to the server like this... https://github.com/NREL/alfalfa/blob/develop/web/server/resolvers.js#L133